.github/workflows/stack-deployment.yaml

```diff
@@ -0,0 +1,48 @@
+name: Stack Deployment
+
+on:
+  workflow_dispatch:
+
+permissions:
+  id-token: write
+  contents: read
+
+jobs:
+  deploy:
+    name: Deploy Stack
+    runs-on: ubuntu-latest
+    env:
+      AWS_REGION: ${{ secrets.AWS_REGION }}
+    steps:
+      - name: Checkout repository
+        uses: actions/checkout@v4
+
+      - name: Setup AWS credentials
+        uses: aws-actions/configure-aws-credentials@v4
+        with:
+          role-to-assume: ${{ secrets.AWS_CDK_ROLE_ARN }}
+          role-session-name: js-assertion-cheat-sheet-stack-deployment
+          aws-region: ${{ secrets.AWS_REGION }}
+
+      - name: Setup Node.js
+        uses: actions/setup-node@v4
+        with:
+          node-version: lts/*
+
+      - name: Install dependencies
+        working-directory: cdk
+        run: npm install
+
+      - name: Deploy stack
+        working-directory: cdk
+        run: npm run deploy -- --require-approval never --outputs-file ./cdk-outputs.json
+
+      - name: Store CDK output
+        run: |
+          cat ./cdk/cdk-outputs.json | jq '.JsAssertionCheatSheet' >> cdk-output.json
+
+      - name: Upload CDK output as artifact
+        uses: actions/upload-artifact@v4
+        with:
+          name: cdk-output
+          path: cdk-output.json
```

Comment:
Would it be good to explicitly set `retention-days`? Currently, the default is 90 days (from repository settings).

Reply:
It depends on how often we would be redeploying the stack. Also, at the moment feature branch deployment/cleanup is also depending on this artifact, so it means that whenever this artifact gets deleted, we will have to execute the stack deployment pipeline again, to be able to deploy/cleanup feature branches.

However, having to execute stack deployment pipeline just because the artifact has expired does not seem like a favorable solution for me. Let me think about this.